### ISSUE TITLE
No interrupt Braille reading when cumputer pass in sleep mode

### ISSUE THREAD

**Issue by CBC33000** [nvaccess/nvda#9175]
Hello,
when I read a text with my Braille display and the cumputer pass in slepp mode, the Braille display post "desktop window".
It is possible to add a feature to prevent cumputer to pass in sleep mode when the Braille display is scrolling ?
Thank's.

**Comment by DrSooom** [comment]
@CBC33000: Please provide us more information by completely filing out the gut report draft. And also tell us your used braille display (including the firmware version) and if you already copied your NVDA configuration for the logon screen (via the General NVDA Settings, admin privileges required). Or in other words: Does your braille display shows different contents on the logon screen after booting your system while moving around there? Thanks.

**Comment by leonardder** [comment]
Hmm, this is an interesting issue. I'm currently not sure when Windows resets the counter that counts up to the moment that sleep mode is activated. If there is an easy way to reset this, it is probably pretty easy to implement this for all braille display drivers.

**Comment by leonardder** [comment]
This turns out to be pretty easy. We can call [SetThreadExecutionState](https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-setthreadexecutionstate) as part of the execution of a braille display gesture, providing it the ES_SYSTEM_REQUIRED flag. This will reset the system idle timer.

**Comment by CBC33000** [comment]
Sleep mode doesn't occures but I noticed screen saver occures so braille reading is interupted. Should I open another issue for that last point?

**Comment by leonardder** [comment]
Unfortunately, this doesn't cover screen saver indeed. However, I believe that everyone should be able to disable screen savers, while disabling of auto sleep is often blocked by group policies.

**Comment by pzajda** [comment]
A suggestion: Why not calling SetThreadExecutionState with the flag ES_DISPLAY_REQUIRED?
As for multimedia applications.

**Comment by DrSooom** [comment]
Screensavers make no sense for LCD monitors. (OLED would be a different topic, but big OLED PC monitors are still quite rare.) And if you want to use a screensaver for getting a password prompt to close it again, it's better to press WIN+L to lock the PC instead.

**Comment by pzajda** [comment]
It is better but unfortunately some people prefer to have this little security practice (screen saver or simply automatic lock screen) to be sure their session will be locked without making the computer in sleep mode.
If you forget to press Windows+l, it is a good thing to have the session locked automatically... Except when reading a very long text in braille.
Point of view about this question may vary for each user so it might be good to let user being free to choose which choice they want to make when it is possible.

**Comment by leonardder** [comment]
The only alternative I see is simulating a key press of an unused key (i.e. f15) or mouse movement without having the mouse actually move. However, where SetThreadExecutionState is a pathway that's officially supported to do this kind of things, simulating input sounds more like a hack/workaround. Before I'd investigate that further, I'd like to have consent from either @michaelDCurran  or @feerrenrut
